### ISSUE TITLE
[BUG] Nil for a counter in MEM group

### ISSUE THREAD

**Issue by ibogosavljevic** [RRZE-HPC/likwid#470]
**Describe the bug**
Sometimes, I get nil for a counter from the MEM group. E.g.

```
+-------------------+------------+
|    Region Info    | HWThread 0 |
+-------------------+------------+
| RDTSC Runtime [s] |  82.901120 |
|     call count    |          1 |
+-------------------+------------+

+-----------------------+---------+--------------+
|         Event         | Counter |  HWThread 0  |
+-----------------------+---------+--------------+
|   INSTR_RETIRED_ANY   |  FIXC0  |  11062130000 |
| CPU_CLK_UNHALTED_CORE |  FIXC1  | 181713400000 |
|  CPU_CLK_UNHALTED_REF |  FIXC2  | 174834500000 |
|       DRAM_READS      | MBOX0C1 | 1.844674e+19 |
|      DRAM_WRITES      | MBOX0C2 |     10344340 |
+-----------------------+---------+--------------+

+-----------------------------------+---------------+
|               Metric              |   HWThread 0  |
+-----------------------------------+---------------+
|        Runtime (RDTSC) [s]        |       82.9011 |
|        Runtime unhalted [s]       |       86.0392 |
|            Clock [MHz]            |     2195.0792 |
|                CPI                |       16.4266 |
|  Memory load bandwidth [MBytes/s] |      nil      |
|  Memory load data volume [GBytes] | 1180591360000 |
| Memory evict bandwidth [MBytes/s] |        7.9859 |
| Memory evict data volume [GBytes] |        0.6620 |
|    Memory bandwidth [MBytes/s]    |      nil      |
|    Memory data volume [GBytes]    |  1.180591e+12 |
+-----------------------------------+---------------+
```

**To Reproduce**
* CPU Intel(R) Core(TM) i5-10210U CPU @ 1.60GHz
* Command: likwid-perfctr -C 0 -m -g MEM ./array_binary_tree
* likwid 5.2.1
* Linux Mint 20.2

**Comment by TomTheBear** [comment]
On desktop chips, the memory controllers only provide free-running counters without any overflow indication. It seems the 32 bit counter overflows multiple times in your case. Does it also happen with short-running applications (<10s) ?

Please provide the output again with `likwid-perfctr -C 0 -m -V 3 -g MEM ./array_binary_tree`. Thanks.

**Comment by ibogosavljevic** [comment]
No, it doesn't happen with shorter running times. The first time it happens when the program is running for 13s. Is there any way to workaround it?

**Comment by TomTheBear** [comment]
With the MarkerAPI, you have to read "frequent enough", so maybe move the Markers inside an inner loop. LIKWID does not read the counters under the hood when the MarkerAPI is used.

Without the MarkerAPI, LIKWID reads the counters every 60s, so too long for your case. I'll do some search how to improve the overflow detection or reduce the background read interval value if certain counters are used. The 60s come from the energy readings (RAPL) which could overflow every 60s and also don't have any overflow indication.

**Comment by ibogosavljevic** [comment]
In the case of a 13 second run, the code enter and leaves the segment 32 times, so it is less than a second. Does likwid initializes those counters to zero?

**Comment by ibogosavljevic** [comment]
So essentially they are unusable for any serious measurement taking more than a second :(

**Comment by TomTheBear** [comment]
I found the problem. The fix will be part of the 5.2.2 release.
